testkit/javadsl/src/main/scala/com/lightbend/lagom/javadsl/testkit/ServiceTest.scala

```diff
@@ -341,11 +342,12 @@ object ServiceTest {
         initialBuilder
```

Comment:
All this change looks great. I think we could make one last improvement extracting all the`CassandraLaunch` stuff into a method so it looks like:

```scala
    val finalBuilder =
      if (setup.cassandra) {
        val cassandraPort: Int = runCassandraServer(testName, lifecycle, log)
        initialBuilder
          .configure(CassandraTestConfig.persistenceConfig(testName, cassandraPort))
          .configure("lagom.cluster.join-self", "on")
          .disableModules(DBModule, JdbcPersistenceModule, KafkaClientModule, KafkaBrokerModule)
      } else if (setup.jdbc) {
...
```

in that case the `if/else` tree will be as concise as it can get using the `GuiceApplicationBuilder` API.

Reply:
Good suggestion!